tests/JsonApiSerializer.Test/DeserializationTests/DeserializationCustomConverterTests.cs

```diff
@@ -1,51 +1,97 @@
-﻿using JsonApiSerializer.Test.Models.Timer;
-using Newtonsoft.Json;
-using System;
-using Xunit;
-
-namespace JsonApiSerializer.Test.SerializationTests
-{
-    public class DeserializationCustomConverterTests
-    {
-        [Fact]
-        public void When_custom_convertor_should_use_on_attributes()
-        {
-            var json = @"{
-  ""data"": {
-    ""type"": ""timer"",
-    ""id"": ""root"",
-    ""attributes"": {
-                ""duration"": 42
-    },
-    ""relationships"": {
-                ""subTimer"": {
-                    ""data"": [
-                      {
-            ""id"": ""sub1"",
-                        ""type"": ""timer""
-          }
-        ]
-      }
-    }
-  },
-  ""included"": [
-    {
-      ""type"": ""timer"",
-      ""id"": ""sub1"",
-      ""attributes"": {
-        ""duration"": 142
-      }
-    }
-  ]
-}";
-
-            var settings = new JsonApiSerializerSettings();
-            var timer = JsonConvert.DeserializeObject<Timer>(json, settings);
-            
-            Assert.Equal(TimeSpan.FromSeconds(42), timer.Duration);
-            Assert.Single(timer.SubTimer);
-            Assert.Equal(TimeSpan.FromSeconds(142), timer.SubTimer[0].Duration);
-        }
-    }
-}
-
+﻿using System;
+using System.Collections.Generic;
+using JsonApiSerializer.JsonConverters;
+using JsonApiSerializer.Test.Models.Products;
+using JsonApiSerializer.Test.Models.Timer;
+using JsonApiSerializer.Test.TestUtils;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using Xunit;
+
+namespace JsonApiSerializer.Test.DeserializationTests
+{
+    public class DeserializationCustomConverterTests
+    {
+        private class ProductContentConverter : ResourceObjectConverter
+        {
+            protected override object CreateDefault(JsonObjectContract contract, string type)
+            {
+                switch (type)
+                {
+                    case "videos":
+                        contract.DefaultCreator = () => new Video();

```

Comment:
A bit confused why you reset the DefaultCreator rather than just new'ing up a new object here. Doing this in an actual implementation could easily lead to weird side affects as I think the contract is a single instance for that type that is shared